### ISSUE TITLE
[CI] MachineLearningIT testDeleteExpiredData

### ISSUE THREAD

**Issue by albertzaharovits** [elastic/elasticsearch#54090]
```
./gradlew ':client:rest-high-level:integTestRunner' --tests "org.elasticsearch.client.MachineLearningIT.testDeleteExpiredData" -Dtests.seed=6520A2DC8EEFE538 -Dtests.security.manager=true -Dtests.locale=nl-NL -Dtests.timezone=Europe/Warsaw -Dcompiler.java=13	
  2> org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:46111], URI [/.ml-state/_refresh], status line [HTTP/1.1 404 Not Found]	
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [.ml-state]","resource.type":"index_or_alias","resource.id":".ml-state","index_uuid":"_na_","index":".ml-state"}],"type":"index_not_found_exception","reason":"no such index [.ml-state]","resource.type":"index_or_alias","resource.id":".ml-state","index_uuid":"_na_","index":".ml-state"},"status":404}	
        at __randomizedtesting.SeedInfo.seed([6520A2DC8EEFE538:AD6F8F70E7F3882A]:0)	
        at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)	
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)	
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)	
        at org.elasticsearch.client.ESRestHighLevelClientTestCase.refreshIndexes(ESRestHighLevelClientTestCase.java:244)	
        at org.elasticsearch.client.ESRestHighLevelClientTestCase.searchAll(ESRestHighLevelClientTestCase.java:236)	
        at org.elasticsearch.client.ESRestHighLevelClientTestCase.searchAll(ESRestHighLevelClientTestCase.java:232)	
        at org.elasticsearch.client.MachineLearningIT.testDeleteExpiredData(MachineLearningIT.java:1008)

```

A total of 3 build failures (7.x and master) over the last few days [MachineLearningIT#testDeleteExpiredData](https://gradle-enterprise.elastic.co/scans/tests?failures.failureClassification=non_verification&list.offset=0&list.size=50&list.sortColumn=startTime&list.sortOrder=desc&search.buildToolType=gradle&search.buildToolType=maven&search.startTimeMax=1585049356650&search.startTimeMin=1584444556650&search.tags=CI&search.tags=not:pull-request&tests.container=org.elasticsearch.client.MachineLearningIT&tests.sortField=FAILED&tests.test=testDeleteExpiredData&tests.unstableOnly=true&trends.section=overview&trends.timeResolution=day&viewer.tzOffset=120)

Possible root cause:
```
        » ERROR][o.e.x.i.IndexLifecycleRunner] [integTest-0] policy [ml-size-based-ilm-policy] for index [.ml-state] failed on step [{"phase":"hot","action":"rollover","name":"check-rollover-ready"}]. Moving to ERROR step	
»  java.lang.IllegalArgumentException: index.lifecycle.rollover_alias [.ml-state-write] does not point to index [.ml-state]	
»  	at org.elasticsearch.xpack.core.ilm.WaitForRolloverReadyStep.evaluateCondition(WaitForRolloverReadyStep.java:104) [x-pack-core-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at org.elasticsearch.xpack.ilm.IndexLifecycleRunner.runPeriodicStep(IndexLifecycleRunner.java:157) [x-pack-ilm-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at org.elasticsearch.xpack.ilm.IndexLifecycleService.triggerPolicies(IndexLifecycleService.java:328) [x-pack-ilm-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at org.elasticsearch.xpack.ilm.IndexLifecycleService.triggered(IndexLifecycleService.java:266) [x-pack-ilm-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at org.elasticsearch.xpack.core.scheduler.SchedulerEngine.notifyListeners(SchedulerEngine.java:183) [x-pack-core-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at org.elasticsearch.xpack.core.scheduler.SchedulerEngine$ActiveSchedule.run(SchedulerEngine.java:211) [x-pack-core-7.7.0-SNAPSHOT.jar:7.7.0-SNAPSHOT]	
»  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_241]	
»  	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_241]	
»  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_241]	
»  	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_241]	
»  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_241]	
»  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_241]	
»  	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_241]
```

Not yet muting since relative fail frequency is very low.

**Comment by droberts195** [comment]
This is the same as #54006.  The backport PR for the fix (#54085) is just about to be merged.

**Comment by przemekwitek** [comment]
> This is the same as #54006. The backport PR for the fix (#54085) is just about to be merged.

Update: It's just got merged.
